integration/query_comparision_test.go

```diff
@@ -265,13 +265,247 @@ func TestQueryComparisonGt(t *testing.T) {
+			expectedIDs: []any{"array-three", "null"},
+		},
+	} {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			var actual []bson.D
+			cursor, err := collection.Find(ctx, tc.q, options.Find().SetSort(bson.D{{"_id", 1}}))
+			require.NoError(t, err)
+			err = cursor.All(ctx, &actual)
+			require.NoError(t, err)
+			assert.Equal(t, tc.expectedIDs, collectIDs(t, actual))
+		})
+	}
+}
+
+func TestQueryComparisonLt(t *testing.T) {
+	t.Parallel()
+	providers := []shareddata.Provider{shareddata.Scalars, shareddata.Composites}
+	ctx, collection := setup(t, providers...)
+
+	for name, tc := range map[string]struct {
+		q           bson.D
+		expectedIDs []any
+	}{
+		"LtDouble": {
+			q:           bson.D{{"value", bson.D{{"$lt", 42.13}}}},
+			expectedIDs: []any{"array", "array-three", "double-negative-infinity", "double-negative-zero", "double-smallest", "double-zero", "int32", "int32-min", "int32-zero", "int64", "int64-min", "int64-zero"},
+		},
+		"LtDoubleNegativeInfinity": {
+			q:           bson.D{{"value", bson.D{{"$lt", math.Inf(-1)}}}},
+			expectedIDs: []any{},
+		},
+		"LtDoubleSmallest": {
+			q:           bson.D{{"value", bson.D{{"$lt", math.SmallestNonzeroFloat64}}}},
+			expectedIDs: []any{"double-negative-infinity", "double-negative-zero", "double-zero", "int32-min", "int32-zero", "int64-min", "int64-zero"},
+		},
+
+		"LtString": {
+			q:           bson.D{{"value", bson.D{{"$lt", "goo"}}}},
+			expectedIDs: []any{"array-three", "string", "string-empty"},
+		},
+
+		"LtEmptyString": {
+			q:           bson.D{{"value", bson.D{{"$lt", ""}}}},
+			expectedIDs: []any{},
+		},
+
+		"LtInt32": {
+			q:           bson.D{{"value", bson.D{{"$lt", int32(42)}}}},
+			expectedIDs: []any{"double-negative-infinity", "double-negative-zero", "double-smallest", "double-zero", "int32-min", "int32-zero", "int64-min", "int64-zero"},
+		},
+
+		"LtInt32Min": {
+			q:           bson.D{{"value", bson.D{{"$lt", int32(math.MinInt32)}}}},
+			expectedIDs: []any{"double-negative-infinity", "int64-min"},
+		},
+
+		"LtInt64": {
+			q:           bson.D{{"value", bson.D{{"$lt", int64(42)}}}},
+			expectedIDs: []any{"double-negative-infinity", "double-negative-zero", "double-smallest", "double-zero", "int32-min", "int32-zero", "int64-min", "int64-zero"},
+		},
+
+		"LtInt64Min": {
+			q:           bson.D{{"value", bson.D{{"$lt", int64(math.MinInt64)}}}},
+			expectedIDs: []any{"double-negative-infinity"},
+		},
+
+		"LtDatetime": {
+			q:           bson.D{{"value", bson.D{{"$lt", time.Date(2021, 11, 1, 10, 18, 41, 123000000, time.UTC)}}}},
+			expectedIDs: []any{"datetime-epoch", "datetime-year-min"},
+		},
+
+		"LtTimeStamp": {
+			q:           bson.D{{"value", bson.D{{"$lt", primitive.Timestamp{T: 41, I: 13}}}}},
+			expectedIDs: []any{"timestamp-i"},
+		},
+		"LtNull": {
+			q:           bson.D{{"value", bson.D{{"$lt", nil}}}},
+			expectedIDs: []any{},
+		},
+	} {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
```

Comment:
this can be extracted to a function, WDYT?

Reply:
Yes, it can be extracted to a helper function to DRY things. Then another test might need a slightly tweaked version of that function, so we just one small addition parameter and one small `if`… And then we end up just like we did in `dance` or `handlers_test.go` that tests everything in one giant test with a lot of branching. We want to avoid that.

So here we don't want to DRY things too much. A few lines copied across several tests there are fine.